addons/html_editor/static/tests/link/popover.test.js

```diff
@@ -257,3 +257,98 @@ describe("Link creation by powerbox", () => {
         expect(getContent(el)).toBe('<p>ab<a href="http://test.com">test.com[]</a></p>');
     });
 });
+
+describe("Link formatting in the popover", () => {
+    test("click on link, the link popover should load the current format correctly", async () => {
+        await setupEditor(
+            '<p><a href="http://test.com/" class="btn btn-outline-primary rounded-circle btn-lg">link2[]</a></p>'
+        );
+        await waitFor(".o-we-linkpopover");
+        click(".o_we_edit_link");
+        await animationFrame();
+        await waitFor("#link-preview");
+        const linkPreviewEl = queryOne("#link-preview");
```

Comment:
I think you can just do `const linkPreviewEl = await waitFor("#link-preview");` here.